add doc for tidb_txn_entry_size_limit (#15714)
pingcap/docs: system-variables.md

</CustomContent>

### `tidb_txn_entry_size_limit` <span class="version-mark">New in v7.6.0</span>

- Scope: SESSION | GLOBAL
- Persists to cluster: Yes
- Applies to hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value): No
- Type: Integer
- Default value: `0`
- Range: `[0, 125829120]`
- Unit: Bytes

<CustomContent platform="tidb">

- This variable is used to dynamically modify the TiDB configuration item [`performance.txn-entry-size-limit`](/tidb-configuration-file.md#txn-entry-size-limit-new-in-v50). It controls the size limit of a single row of data in TiDB, which is equivalent to the configuration item. The default value of this variable is `0`, which means that TiDB uses the value of the configuration item `txn-entry-size-limit` by default. When it is set to a non-zero value, the value is used as the value of `txn-entry-size-limit`.

</CustomContent>

> <CustomContent platform="tidb-cloud">

Author: @zyguan Please help confirm if this is also applicable to TiDB Cloud

Reviewer: Yes.

> - This variable is used to dynamically modify the TiDB configuration item [`performance.txn-entry-size-limit`](https://docs.pingcap.com/tidb/stable/tidb-configuration-file#txn-entry-size-limit-new-in-v50). It controls the size limit of a single row of data in TiDB, which is equivalent to the configuration item. The default value of this variable is `0`, which means that TiDB uses the value of the configuration item `txn-entry-size-limit` by default. When it is set to a non-zero value, the value is used as the value of `txn-entry-size-limit`.

Reviewer 2: ```suggestion
- This variable is used to dynamically modify the TiDB configuration item [`performance.txn-entry-size-limit`](https://docs.pingcap.com/tidb/stable/tidb-configuration-file#txn-entry-size-limit-new-in-v50). It limits the size of a single row of data in TiDB, which is equivalent to the configuration item. The default value of this variable is `0`, which means that TiDB uses the value of the configuration item `txn-entry-size-limit` by default. When this variable is set to a non-zero value, `txn-entry-size-limit` is also set to the same value.
```

</CustomContent>

> **Note:**
>

> > Modifying this variable with the SESSION scope only affects the current user session, not the internal TiDB session. This might lead to transaction failure if the entry size of the internal TiDB transaction exceeds the limit of the configuration. Therefore, to dynamically increase the limit, it is recommended that you modify the variable with the GLOBAL scope.

Reviewer 2: ```suggestion
> Modifying this variable with the SESSION scope only affects the current user session, not the internal TiDB session. This might lead to transaction failure if the entry size of an internal TiDB transaction exceeds the limit of the configuration item. Therefore, to dynamically increase the limit, it is recommended that you modify the variable with the GLOBAL scope.
```
